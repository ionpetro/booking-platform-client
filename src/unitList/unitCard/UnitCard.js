export default {
  name: 'UnitCard',
  props: {
    unit: {
      required: true
    }
  },
  /*
    The @hadDiscount returns true if the unit has a discount
    and false if it doesn't.
  */
  computed: {
    hasDiscount() {
      return !(this.unit.price === this.unit.discountedPrice);
    }
  }
};
