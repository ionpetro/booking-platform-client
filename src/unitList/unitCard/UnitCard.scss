.card {
  position: relative;
  width: 400px;
  transition: transform 0.25s ease-in;
  cursor: pointer;

  @media only screen and (max-width: $medium-breakpoint) {
    width: 328px;
  }

  &__image {
    width: 400px;
    height: 200px;
    border-radius: 4px 30px;
    margin-bottom: 125px;
    overflow: hidden;

    img {
      width: 400px;
      height: 200px;
    }
    @media only screen and (max-width: $medium-breakpoint) {
      width: 328px;
      height: 160px;
    }
  }

  &__image--zoom {
    img {
      transform-origin: 50% 65%;
      transition: transform 0.25s ease-in-out;
    }
  }

  &__info {
    position: absolute;
    left: 60px;
    top: 176px;
    background-color: $white;
    height: 126px;
    width: 288px;
    border-radius: 4px;
    padding: 0 30px;
    box-shadow: 0px 4px 20px rgba(83, 83, 83, 0.15);
    transition: transform 0.25s ease-in-out;

    @media only screen and (max-width: $medium-breakpoint) {
      left: 22px;
      top: 140px;
    }

    &__distance {
      color: $gray-dim;
    }
  }

  &__price {
    padding-top: 14px;
    margin: 0;
    &__amount {
      font-weight: 700;
    }

    &__discount {
      color: $red-medium;
      text-decoration: line-through;
      padding-right: 2px;
    }
  }

  &__location {
    padding-top: 10px;
  }

  &__amenities {
    padding-top: 10px;
    display: flex;
    justify-content: flex-start;

    div {
      display: flex;
      span {
        padding: 4px 20px 4px 4px;
      }
    }
  }

  &:hover {
    transform: scale(1.03);

    .card__image {
      box-shadow: 0px 4px 20px rgba(83, 83, 83, 0.15);
    }

    .card__info {
      transform: scale(0.95);
      box-shadow: 0px 4px 10px rgba(83, 83, 83, 0.15);
    }
    .card__image--zoom {
      img {
        transform: scale(1.08);
      }
    }
  }
}
