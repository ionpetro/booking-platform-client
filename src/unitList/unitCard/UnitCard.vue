<template>
  <div class="card">
    <div class="card__image card__image--zoom">
      <img :src="unit.image" alt="Unit Image" />
    </div>
    <div class="card__info">
      <div class="card__price">
        <span class="body--small card__price__discount" v-if="hasDiscount">{{ unit.price }}</span>
        <span class="body--large card__price__amount"
          >{{ unit.discountedPrice }}
          <img src="@/assets/icons/Bitcoin.svg" alt="Bitcoin icon" />
        </span>
      </div>
      <div class="card__location">
        {{ unit.crater }}, {{ unit.address }}
        <span class="card__info__distance">â€¢ {{ unit.distance }}pt from center</span>
      </div>
      <div class="card__characteristics">
        <div v-if="unit.capsules">
          <img src="@/assets/icons/Capsules.svg" alt="Capsules" /><span> {{ unit.capsules }}</span>
        </div>
        <div v-if="unit.sterilizers">
          <img src="@/assets/icons/Sterilizers.svg" alt="sterilizers" /><span>
            {{ unit.sterilizers }}</span
          >
        </div>
        <div v-if="unit.size">
          <img src="@/assets/icons/Size.svg" alt="Size" />
          <span> {{ unit.size }}mf</span>
        </div>
      </div>
    </div>
  </div>
</template>

<script src="./UnitCard.js"></script>

<style lang="scss" src="./UnitCard.scss" scoped></style>
