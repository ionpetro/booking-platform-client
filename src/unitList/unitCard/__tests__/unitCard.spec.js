import {mount} from '@vue/test-utils';
import UnitCard from '../UnitCard.vue';

describe('UnitCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(UnitCard, {
      propsData: {
        unit: {
          id: 1,
          image: 'image',
          crater: 'name',
          distance: 1,
          facilities: {
            capsules: 1,
            sterilizers: 1,
            size: 1
          },
          price: 1,
          address: 1,
          discount: true
        }
      }
    });
  });

  describe('when loaded', () => {
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
