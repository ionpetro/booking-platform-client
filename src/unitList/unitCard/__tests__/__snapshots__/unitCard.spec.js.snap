// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitCard when loaded renders correctly 1`] = `
<div
  class="card"
>
  <div
    class="card__image card__image--zoom"
  >
    <img
      alt="Unit Image"
      src="image"
    />
  </div>
   
  <div
    class="card__info"
  >
    <div
      class="card__price"
    >
      <span
        class="body--small card__price__discount"
      >
        1
      </span>
       
      <span
        class="body--large card__price__amount"
      >
        
        
        <img
          alt="Bitcoin icon"
          src="@/assets/icons/Bitcoin.svg"
        />
      </span>
    </div>
     
    <div
      class="card__location"
    >
      
      name, 1
      
      <span
        class="card__info__distance"
      >
        â€¢ 1pt from center
      </span>
    </div>
     
    <div
      class="card__amenities"
    >
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;
